django-stubs/test/client.pyi

```diff
@@ -28,45 +44,68 @@ class FakePayload:
     def read(self, num_bytes: int = ...) -> bytes: ...
     def write(self, content: Union[bytes, str]) -> None: ...
 
+_T = TypeVar("_T")
+
+def closing_iterator_wrapper(iterable: Iterable[_T], close: Callable[[], Any]) -> Iterator[_T]: ...
+def conditional_content_removal(request: HttpRequest, response: HttpResponseBase) -> HttpResponseBase: ...
+
 class ClientHandler(BaseHandler):
     enforce_csrf_checks: bool = ...
     def __init__(self, enforce_csrf_checks: bool = ..., *args: Any, **kwargs: Any) -> None: ...
     def __call__(self, environ: Dict[str, Any]) -> HttpResponseBase: ...
 
+class AsyncClientHandler(BaseHandler):
+    enforce_csrf_checks: bool = ...
+    def __init__(self, enforce_csrf_checks: bool = ..., *args: Any, **kwargs: Any) -> None: ...
+    async def __call__(self, scope: Dict[str, Any]) -> HttpResponseBase: ...
+
 def encode_multipart(boundary: str, data: Dict[str, Any]) -> bytes: ...
 def encode_file(boundary: str, key: str, file: Any) -> List[bytes]: ...
 
-class RequestFactory:
+# fake to distinguish WSGIRequest and ASGIRequest
+
+_R = TypeVar("_R", bound=HttpRequest)
+
+class _MonkeyPatchedHttpResponseBase(Generic[_R], HttpResponseBase):
```

Comment:
No: it can be `FileResponse` that inherits from `HttpResponseBase`.

Reply:
Looks like the `__iter__` and `getvalue()` APIs are "the right way" to access the response content in a universal manner -- both `StreamingHttpResponse` and `HttpResponse` define these methods.

But since these methods aren't defined on the level of `HttpResponseBase` either, this is still an issue.

Should we add `__iter__` and `getvalue()` to `HttpResponseBase` since presumably all subclasses are expected to implement these methods?